website_product/controllers/main.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2018-2021 ForgeFlow S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+from odoo import http
+from odoo.http import request
+
+
+class WebsiteProductPage(http.Controller):
+    @http.route(
+        ['/catalog/product/<model("product.template"):product>'],
```

Comment:
You should add a filter on website_product_published here to prevent showing 'non-catalog' products when creating menu items etc.
![image](https://user-images.githubusercontent.com/10028499/157889086-97d3db9d-bffb-4135-b345-2b7da1e08aee.png)